src/doc/unstable-book/src/compiler-flags/sanitizer.md

```diff
@@ -865,6 +867,58 @@ WARNING: ThreadSanitizer: data race (pid=10574)
   Location is global 'example::A::h43ac149ddf992709' of size 8 at 0x5632dfe3d030 (example+0x000000bd9030)
 ```
 
+# RealtimeSanitizer
+RealtimeSanitizer detects non-deterministic execution time calls in real-time contexts.
+Function marked with the `#[sanitize(realtime = "nonblocking")]` attribute are considered real-time functions.
+When RTSan detects a call to a function with non-deterministic execution time, like `malloc` or `free`
+while in a real-time context, it reports an error.
+
+Besides "nonblocking" the attribute can also be used with "blocking" and "caller".
+- "blocking" marks the function as having a non-deterministic execution time. When reaching such
```

Comment:
Possibly:

"blocking" allows programmers to mark a function as having non-deterministic execution time, but not necessarily calling a system call. When calling a "blocking" function in a real-time context, a violation will be reported. A classic use case of this is a userland spinlock.


(adding an example, indicating that it's a way to show intent of what the API does)